haxe/ui/core/Component.hx

```diff
@@ -46,23 +49,24 @@ class DeferredBindingInfo {
 @:autoBuild(haxe.ui.macros.Macros.buildBindings())
 @:build(haxe.ui.macros.Macros.addClonable())
 @:autoBuild(haxe.ui.macros.Macros.addClonable())
-class Component extends ComponentBase implements IComponentBase implements IClonable<Component> {
+class Component extends ComponentBase implements IComponentBase implements IValidating implements IClonable<Component> {
     public function new() {
         super();
 
         #if flash
-        addClass("flash");
+        addClass("flash", false);
         #end
         #if html5
-        addClass("html5");
+        addClass("html5", false);
         #end
-        addClass(Backend.id);
+        addClass(Backend.id, false);
 
         var parts:Array<String> = Type.getClassName(Type.getClass(this)).split(".");
         var className:String = parts[parts.length - 1].toLowerCase();
         addClass(className, false);
 
         // we dont want to actually apply the classes, just find out if native is there or not
+        //TODO - we could include the initialization in the validate method
         var s = Toolkit.styleSheet.applyClasses(this, false);
         if (s.native != null && hasNativeEntry == true) {
```

Comment:
Just an fyi: all this is VERY important for the native (hxwidgets) and hybrid stuff in haxeui2. I havent tested, but just making a note.